packages/eslint-plugin/src/index.ts

```diff
@@ -17,23 +17,31 @@ import type { TSESLint } from "@typescript-eslint/utils";
 
 import rules from "./rules";
 
-const blueprintPlugin = { configs: { recommended: {} }, rules };
+type ConfigName = "recommended";
 
-// Assign the config here so that we can reference blueprintPlugin.
-const configs: { [c in keyof (typeof blueprintPlugin)["configs"]]: TSESLint.FlatConfig.Config } = {
-    /**
-     * Enables all Blueprint-specific lint rules defined in this package.
-     */
-    recommended: {
-        plugins: { "@blueprintjs": blueprintPlugin },
-        rules: {
-            "@blueprintjs/classes-constants": "error",
-            "@blueprintjs/html-components": "error",
-            "@blueprintjs/no-deprecated-components": "error",
-            "@blueprintjs/no-deprecated-type-references": "error",
-        },
+const blueprintPlugin = {
+    configs: { recommended: {} } as Record<ConfigName, TSESLint.ClassicConfig.Config>,
+    flatConfigs: { recommended: {} } as Record<ConfigName, TSESLint.FlatConfig.Config>,
```

Comment:
Renaming `configs` to `flatConfigs` here so that devs that have `extends: ["plugin:@blueprintjs/recommended"]` in their `.eslintrc` file still refer to the legacy configuration (and hence avoiding a breaking change).